staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go

```diff
@@ -58,13 +60,19 @@ func NamedCheck(name string, check func(r *http.Request) error) HealthzChecker {
 	return &healthzCheck{name, check}
 }
 
-// InstallHandler registers a handler for health checking on the path "/healthz" to mux.
+// InstallHandler registers handlers for health checking on the path
+// "/healthz" to mux. *All handlers* for mux must be specified in
+// exactly one call to InstallHandler. Calling InstallHandler more
+// than once for the same mux will result in a panic.
 func InstallHandler(mux mux, checks ...HealthzChecker) {
 	if len(checks) == 0 {
+		glog.Info("No default health checks specified. Installing the ping handler.")
 		checks = []HealthzChecker{PingHealthz}
 	}
+	glog.Infof("Installing %v healthz checkers", len(checks))
 	mux.Handle("/healthz", handleRootHealthz(checks...))
 	for _, check := range checks {
+		glog.Infof("Installing %q healthz checker", check.Name())
```

Comment:
This ought to run exactly once at startup. Multiple calls to `InstallHandler()` will fail after the first.

The `name` is a user provided string, so we have to have some faith that people will not do something weird. Do you have an alternate suggestion or should I change it to v(4) now?